core/sail/lmdb/src/test/java/org/eclipse/rdf4j/sail/lmdb/benchmark/QueryBenchmark.java

```diff
@@ -151,100 +160,140 @@ public long complexQuery() {
 	}
 
 	@Benchmark
-	public long distinctPredicatesQuery() {
+	public long pathExpressionQuery1() {
 
 		try (SailRepositoryConnection connection = repository.getConnection()) {
 			return connection
-					.prepareTupleQuery(query5)
+					.prepareTupleQuery(query7_pathexpression1)
 					.evaluate()
 					.stream()
 					.count();
```

Comment:
I believe I changed this code a bit in the memory store. The .count() method doesn't call close. I made a small helper method to simplify the code a bit:

```
	private static long count(TupleQueryResult evaluate) {
		try (Stream<BindingSet> stream = evaluate.stream()) {
			return stream.count();
		}
	}
```

Then you can use it like this:

```
@Benchmark
	public long complexQuery() {
		try (SailRepositoryConnection connection = repository.getConnection()) {
			return count(connection
					.prepareTupleQuery(query4)
					.evaluate()
			);
		}
	}
```

Reply:
I've fixed this.